test/index.js

```diff
@@ -0,0 +1,353 @@
+    ],
+    symmetry: [
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+    ],
+    transitivity: [
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+    ],
+  });
+});
+
+suite ('Ord laws', () => {
+  testLaws (laws.Ord) ({
+    totality: [
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+    ],
+    antisymmetry: [
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+    ],
+    transitivity: [
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+      ConstantArb (Number) (NumberArb),
+    ],
+  });
+});
+
+suite ('Semigroup laws', () => {
+  testLaws (laws.Semigroup (Z.equals)) ({
+    associativity: [
+      ConstantArb (String) (jsc.string),
+      ConstantArb (String) (jsc.string),
+      ConstantArb (String) (jsc.string),
+    ],
+  });
+});
+
+suite ('Functor laws', () => {
+  testLaws (laws.Functor (Z.equals)) ({
+    identity: [
+      ConstantArb (String) (jsc.string),
+    ],
+    composition: [
+      ConstantArb (String) (jsc.string),
+      jsc.constant (Math.sqrt),
+      jsc.constant (Math.abs),
+    ],
+  });
+});
+
+suite ('Bifunctor laws', () => {
+  testLaws (laws.Bifunctor (Z.equals)) ({
+    identity: [
+      ConstantArb (String) (jsc.string),
+    ],
+    composition: [
+      ConstantArb (String) (jsc.string),
+      jsc.constant (Math.sqrt),
+      jsc.constant (s => s.length),
+      jsc.constant (Math.sqrt),
+      jsc.constant (Math.abs),
+    ],
+  });
+});
+
+suite ('Apply laws', () => {
+  testLaws (laws.Apply (Z.equals)) ({
+    composition: [
+      ConstantArb (String) (jsc.string),
+      ConstantArb (String) (jsc.string),
+      ConstantArb (String) (jsc.string),
+    ],
+  });
+});
+
+suite ('Applicative laws', () => {
+  testLaws (laws.Applicative (Z.equals, Constant (Array))) ({
+    identity: [
+      ConstantArb (Array) (jsc.array (jsc.number)),
+    ],
+    homomorphism: [
+      jsc.constant (Math.abs),
+      jsc.number,
+    ],
+    interchange: [
+      ConstantArb (Array) (jsc.array (jsc.number)),
+      jsc.string,
+    ],
+  });
+});
+
+suite ('Foldable laws', () => {
+  testLaws (laws.Foldable (Z.equals)) ({
+    associativity: [
+      jsc.constant (s => x => s + s),
+      jsc.string,
+      ConstantArb (Number) (jsc.number),
+    ],
+  });
+});
+
+suite ('Traversable laws', () => {
+  testLaws (laws.Traversable (Z.equals)) ({
+    naturality: [
+      jsc.constant (Array),
+      jsc.constant (Identity),
+      jsc.constant (xs => Identity (xs[0])),
+      ConstantArb (Array) (NonEmpty (jsc.array (jsc.number))),
+    ],
+    identity: [
+      jsc.constant (Array),
+      ConstantArb (String) (jsc.string),
+    ],
+    composition: [
+      jsc.constant (Array),
+      jsc.constant (Identity),
+      ConstantArb (Array) (jsc.array (IdentityArb (jsc.string))),
+    ],
+  });
+});
```

Comment:
@Avaq, please pay particularly close attention to the traversable laws (which I still don't fully understand).

Reply:
I'm not too familiar with the Traversable type class myself, as none of the types I maintain implement it. From what I can tell, nothing seems off about this test.

What does stand out to me in each of these tests is the level of specialization you're testing with, loosely indicated by how often a `jsc.constant` is used in the test parameters.

Ideally, every test would pass with an `AnyConstantArb` defined in terms of itself to include nesting (because the fantasy laws are often broken once nesting comes into play):

```js
const {AnyConstantArb} = jsc.letrec(tie => ({
  AnyConstantArb: jsc.oneof(
    ConstantArb(Number)(jsc.number),
    ConstantArb(String)(jsc.string),
    //...some more?
    ConstantArb(Constant)(tie('AnyConstantArb'))
  )
}))
```

We can also define an arbitrary function `a -> b`:

```js
const ABArb = jsc.oneof(
  a => ({a}),
  b => ({b}),
  c => [c],
  d => Constant['fantasy-land/of'](d)
);
```

Now the Functor composition parameters, for example, could be replaced with:

```js
[AnyConstantArb, AnyFunctionArb, AnyFunctionArb]
```

And you'd be testing for a much larger set of edge-cases.

Reply:
> `ConstantArb(Constant)(tie('AnyConstantArb'))`

I see the problem here.

Reply:
To fix that, I guess we'd have to define some kind of `TypeRepWithArbPairArb` (an arbitrary pair like `[String, jsc.string]`, for example) to generate valid ConstantArbs to be used in recursive definitions of AnyConstantArb. Maybe a little over-ambitious for now. 

I took the approach above from recent changes I made to the Fluture tests, and I figured they could be applied to other types as well. `Constant` is an odd exception because it's a specialized type by nature.

Reply:
I love this idea, Aldwin. Although it's not easy to apply here, we should certainly adopt it for our other ADTs. If you're interested in working on this, I'd love to see a pull request for one of the other projects. :D